### DISCUSSION TITLE
Does Candle have an implementation of the classification head for BERT or should I implement it myself?

### DISCUSSION THREAD

**Discussion by manifest** [huggingface/safetensors#425] [Q&A] [answered]
I've got model parameters in safetensor format by fine-tuning a BERT model with the classification head using [Transformers](https://github.com/huggingface/transformers) library (i.e. using `AutoModelForSequenceClassification` model class). Now, I want to perform inference using [Cande](https://github.com/huggingface/candle/) library.

Does Candle have an implementation of the classification head for BERT or should I implement it myself?

**Comment by manifest** [comment]
I've implemented a classification head for `DistilBertModel` similarly to `DistilBertForSequenceClassification` implementation of Transformers library. Does it make sense to contribute the code to `candle_transformers` crate?

**Reply by Narsil** [reply, marked as answer]
You can definitely share the example, it'll help others for sure.

As for the inclusion within candle-transformers, I'm not sure we have taken a stance on implementing all the potential heads in transformers. So we might not merge it or not soon. Currently bert is mostly used (to the best of my knowledge) for RAG implementations, which is why it has the current form. Adding the classification in client code seems the most reasonable for me at the present.

**Comment by gilljon** [comment]
Mind sharing the implementation? Trying to understand how to use BERT for ranking :)

**Reply by ghost** [reply]
Here's a functional code

https://github.com/e1732a364fed/candle-token-classification
